enforcer-rules/src/test/java/org/apache/maven/enforcer/rules/files/TestRequireFilesSize.java

```diff
@@ -116,7 +122,7 @@ void testEmptyFileList() throws EnforcerRuleException, IOException {
     void testFileDoesNotExist() throws IOException {
         File f = File.createTempFile("junit", null, temporaryFolder);
         f.delete();
-        assertFalse(f.exists());
+        assumeFalse(f.exists());
```

Comment:
It will change test logic ... at all 
When file will be not deleted .. test will be marked as skipped not failed

Is it intentional?

Reply:
Yes, we don't want to test that the JDK's delete method works as expected, and if the code under test isn't even executed then the test has been skipped, not failed. This case inspired me to write https://cafe.elharo.com/testing/assume-vs-assert/